webui/react/src/App.tsx

```diff
@@ -36,8 +36,13 @@ const AppView: React.FC = () => {
   useEffect(() => {
     setupAnalytics(info);
 
-    // Check to make sure the WebUI version matches the platform version.
-    if (info.version !== process.env.VERSION) {
+    /*
+     * Check to make sure the WebUI version matches the platform version.
+     * If the release is a release candidate or a dev tag, skip this notification.
+     * For example, check will run for `0.15.5` but not `0.15.5.dev3`.
+     */
+    const versionChunks = info.version.split('.') || [];
+    if (versionChunks.length === 3 && info.version !== process.env.VERSION) {
       const btn = <Button type="primary" onClick={refreshPage}>Update Now</Button>;
       const message = 'New WebUI Version';
```

Comment:
sorta unrelated: could we run into a case where the UI is newer than the cluster version? 1) we try to detect older or newer 2) we generalize the messaging?